zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -355,8 +355,7 @@ public long getLastLoggedZxid() {
         // the highest zxid
         long zxid = maxLog;
         TxnIterator itr = null;
-        try {
-            FileTxnLog txn = new FileTxnLog(logDir);
+        try (FileTxnLog txn = new FileTxnLog(logDir)) {
             itr = txn.read(maxLog);
```

Comment:
To get equivalent behavior, I think you want to do something like:

```java
try (FileTxnLog txn = new FileTxnLog(logDir); TxnIterator itr = txn.read(maxLog)) {
  // ...
}
```

Reply:
I've updated the PR.

Reply:
I've created https://github.com/apache/zookeeper/pull/1507 and https://issues.apache.org/jira/browse/ZOOKEEPER-3971